ci/Dockerfile

```diff
@@ -0,0 +1,189 @@
+        && \
+    rm -rf ${BOOST_ARCHIVE} && \
+    rm -rf ${BOOST_BUILD_DIR} && \
+    rm -rf ${BOOST_SOURCE_DIR}
+ENV BOOST_ROOT=${INSTALL_DIR}/boost
+
+# Install Trilinos
+RUN export TRILINOS_VERSION=12-12-1 && \
+    export TRILINOS_URL=https://github.com/trilinos/Trilinos/archive/trilinos-release-${TRILINOS_VERSION}.tar.gz && \
+    export TRILINOS_ARCHIVE=${ARCHIVE_DIR}/trilinos.tar.gz && \
+    export TRILINOS_SOURCE_DIR=${SOURCE_DIR}/trilinos && \
+    export TRILINOS_BUILD_DIR=${BUILD_DIR}/trilinos && \
+    export TRILINOS_INSTALL_DIR=${INSTALL_DIR}/trilinos && \
+    wget --quiet ${TRILINOS_URL} --output-document=${TRILINOS_ARCHIVE} && \
+    mkdir -p ${TRILINOS_SOURCE_DIR} && \
+    tar -xf ${TRILINOS_ARCHIVE} -C ${TRILINOS_SOURCE_DIR} --strip-components=1 && \
+    mkdir ${TRILINOS_BUILD_DIR} && \
+    cd ${TRILINOS_BUILD_DIR} && \
+    cmake -DCMAKE_BUILD_TYPE=RELEASE -DBUILD_SHARED_LIBS=ON \
+        -DTPL_ENABLE_MPI=ON \
+        -DTPL_ENABLE_BLAS=ON \
+        -DTPL_ENABLE_LAPACK=ON \
+        -DTPL_ENABLE_Boost=ON \
+        -DBoost_INCLUDE_DIRS=${BOOST_ROOT}/include \
+        -DTPL_ENABLE_BoostLib=ON \
+        -DBoostLib_INCLUDE_DIRS=${BOOST_ROOT}/include \
+        -DBoostLib_LIBRARY_DIRS=${BOOST_ROOT}/lib \
+        -DTrilinos_ENABLE_ALL_PACKAGES=OFF \
+        -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES=OFF \
+        -DTrilinos_ENABLE_TESTS=OFF \
+        -DTrilinos_ENABLE_EXAMPLES=OFF \
+        -DTrilinos_ENABLE_Amesos=ON \
+        -DTrilinos_ENABLE_AztecOO=ON \
+        -DTrilinos_ENABLE_Epetra=ON \
+        -DTrilinos_ENABLE_EpetraExt=ON \
+        -DTrilinos_ENABLE_Ifpack=ON \
+        -DTrilinos_ENABLE_ML=ON \
+        -DTrilinos_ENABLE_MueLu=ON \
+        -DTrilinos_ENABLE_Sacado=ON \
+        -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
+        -DCMAKE_INSTALL_PREFIX=${TRILINOS_INSTALL_DIR} \
+        ${TRILINOS_SOURCE_DIR} && \
+    make -j${N_PROCS} install && \
+    rm -rf ${TRILINOS_ARCHIVE} && \
+    rm -rf ${TRILINOS_BUILD_DIR} && \
+    rm -rf ${TRILINOS_SOURCE_DIR}
+ENV TRILINOS_DIR=${INSTALL_DIR}/trilinos
+
+# Install p4est
+RUN export P4EST_VERSION=1.1 && \
+    export P4EST_URL=http://p4est.github.io/release/p4est-${P4EST_VERSION}.tar.gz && \
+    export P4EST_ARCHIVE=${ARCHIVE_DIR}/p4est-${P4EST_VERSION}.tar.gz && \
+    export P4EST_SOURCE_DIR=${SOURCE_DIR}/p4est && \
+    export P4EST_BUILD_DIR=${BUILD_DIR}/p4est && \
+    export P4EST_INSTALL_DIR=${INSTALL_DIR}/p4est && \
+    wget --quiet ${P4EST_URL} --output-document=${P4EST_ARCHIVE} && \
+    mkdir -p ${P4EST_SOURCE_DIR} && \
+    cd ${P4EST_SOURCE_DIR} && \
+    wget --quiet https://www.dealii.org/8.5.0/external-libs/p4est-setup.sh && \
+    bash ./p4est-setup.sh ${P4EST_ARCHIVE} ${P4EST_INSTALL_DIR} && \
+    rm -rf ${P4EST_ARCHIVE} && \
+    rm -rf ${P4EST_BUILD_DIR} && \
+    rm -rf ${P4EST_SOURCE_DIR}
+ENV P4EST_DIR=${INSTALL_DIR}/p4est
+
+# Install deal.II
+RUN export DEAL_II_URL=https://github.com/dealii/dealii.git && \
```

Comment:
Is it a good idea to run on the tip of deallii?

Reply:
That will be necessary for CUDA. CUDA support has not been release yet. I could use a hash if you think it makes sense

Reply:
Yes, I think we should freeze the version we test against. Otherwise, we are susceptible to dealii breakage.